modules/launcher/src/main/java/org/mule/module/launcher/artifact/AbstractArtifactClassLoader.java

```diff
@@ -60,9 +69,62 @@ public void dispose()
                 logger.error(e);
             }
         }
+
+        LeakCleaner leakCleaner = getLeakCleanerInstance();
+        if (leakCleaner != null)
+        {
+            leakCleaner.clean();
+        }
+
         super.dispose();
     }
 
+    public String getLeakCleanerClassName()
+    {
+        return StringUtils.defaultString(leakCleanerClassName, DEFAULT_LEAK_CLEANER_CLASS);
```

Comment:
Using it this way you ensure that setting leakCleanerClassName to null with the setter won't remove the leak cleaning, but i'm not sure if we have to take care of that, so I think it's ok to set the default value to the field definition.